backend/controllers/eventsController.js

```diff
@@ -1,6 +1,122 @@
+      }
+    }
+    // When a transaction has been approved. EventType=vehicle_approved
+    else if (req.method=='PUT') {
+      const vehicle = await Vehicle.find({ sale: req.params.saleId });
+      if (!!req.body.date_approved) {
+        body = {
+          event_type: 'vehicle_approved',
+          dealership: vehicle[0].dealership,
+          vehicle: vehicle[0]._id,
+          title: 'Transaction Approved',
+          description: `A sale request has been approved for the vehicle with vin: ${vehicle[0].vin} by ${req.user.first_name} ${req.user.last_name}`,
+        }
+      }
+      else if (!!req.body.sale_amount) {
+        body = {
+          event_type: 'transaction_modified',
+          dealership: vehicle[0].dealership,
+          vehicle: vehicle[0]._id,
+          title: 'Transaction Modified',
+          description: `A transaction has been modified for the vehicle with vin: ${vehicle[0].vin} by ${req.user.first_name} ${req.user.last_name}`,
+        }
+      }
+    }
+  }
+
+  //Vehicle API Events
+  else if (req.originalUrl.indexOf('/api/v1/inventory/vehicle/') > - 1) {
+    const vehicle = await Vehicle.findById(req.params.vehicleId);
+    if (req.method == 'DELETE') {
+      body = {
+        event_type: 'vehicle_deleted',
+        dealership: vehicle.dealership,
+        vehicle: vehicle._id,
+        title: 'Vehicle Deleted',
+        description: `A vehicle with vin: ${vehicle.vin} has been deleted by ${req.user.first_name} ${req.user.last_name}`,
+      }
+    }
+    else if (req.method == 'PUT') {
+      if (req.body.delivered) {
+        body = {
+          event_type: 'vehicle_delivered',
+          dealership: vehicle.dealership,
+          vehicle: vehicle._id,
+          title: 'Vehicle Delivered',
+          description: `A vehicle with vin: ${vehicle.vin} has been marked delivered by ${req.user.first_name} ${req.user.last_name}`,
+        }
+      }
+      //Change the if condition later
+      else if (vehicle.missing != req.body.missing) {
+        if (req.body.missing) {
+          body = {
+            event_type: 'vehicle_missing',
+            dealership: vehicle.dealership,
+            vehicle: vehicle._id,
+            title: 'Vehicle Missing',
+            description: `A vehicle with vin: ${vehicle.vin} has been marked missing by ${req.user.first_name} ${req.user.last_name}`,
+          }
+        }
+        else {
+          body = {
+            event_type: 'vehicle_found',
+            dealership: vehicle.dealership,
+            vehicle: vehicle._id,
+            title: 'Vehicle Found',
+            description: `A vehicle with vin: ${vehicle.vin} has been marked found by ${req.user.first_name} ${req.user.last_name}`,
+          }
+        }
+      }
+    }
+  }
+
+  //if body is defined, then create the event
+  if (Object.keys(body).length !== 0) {
+    const event = await Event.create(body);
```

Comment:
No need to create the variable since it is never called after, can just call the function

Reply:
Nice catch. I have deleted that variable and only kept the function call in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/64/commits/91a138c850c326f13efe84f0bc25e9fe6d8d1929)